zbar/qrcode/qrdectxt.c

```diff
@@ -92,7 +92,11 @@ int qr_code_data_list_extract_text(const qr_code_data_list *_qrlist,
   for(i=0;i<nqrdata;i++)if(!mark[i]){
     const qr_code_data       *qrdataj;
     const qr_code_data_entry *entry;
+#ifdef ALL_ENCODINGS
     iconv_t                   enc_list[4];
+#else
+    iconv_t                   enc_list[0];
```

Comment:
You are allocating an empty list on the stack... i.e. 0 bytes. luckily followed by reserving then space for an ```iconv_t```...